aviatrix/utils.go

```diff
@@ -224,7 +224,7 @@ func DiffSuppressFuncNatInterface(k, old, new string, d *schema.ResourceData) bo
 	// Check if the number of snat policies has not changed so that interface can be set when a policy is added.
 	// Without this check, the value for interface will be suppressed and interface = "" will
 	// be passed to the API even if interface = "eth0" in the configuration.
-	if !d.HasChange("snat_policy.#") && !(connection == "" || connection == "None") {
+	if !d.HasChange("snat_policy.#") && !d.HasChange("dnat_policy.#") && !(connection == "" || connection == "None") {
```

Comment:
Probably need to update the comment too since it seems specific to snat.

Would be nice to link to terraform docs about what `.#` means in a key. I couldn't find it and I assume it is special in some way?

Reply:
Good point. I updated the comment. Also reference the bug that was filed to remove this check entirely since we shouldn't need it.  As for the `.#` notation. I couldn't find a good reference for it in documentation but I see it used in the code: https://github.com/hashicorp/terraform/blob/9cf5350590ab25e8f2b51a69f9136f11d97369a0/helper/schema/resource_diff.go#L86

I can dig a little deeper though if you'd like.